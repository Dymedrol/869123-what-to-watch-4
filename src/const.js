const Genres = {
  ALL: `All genres`,
  COMEDIES: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMA: `Drama`,
  HORROR: `Horror`,
  KIDS_FAMILY: `Kids & Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci-Fi`,
  THRILLERS: `Thrillers`,
};

const Tabs = [
  `Overview`,
  `Details`,
  `Reviews`
];

const MovieListStep = {
  MAIN: 8,
  MOVIEPAGE: 4,
};

const videoPlayerModes = {
  SMALL: `small`,
  FULLSCREEN: `fullscreen`,
};

export {Genres, Tabs, MovieListStep, videoPlayerModes};
