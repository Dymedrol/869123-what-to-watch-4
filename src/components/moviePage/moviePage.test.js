import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {BrowserRouter} from 'react-router-dom';

import MoviePage from "./moviePage.jsx";
import NameSpace from "../../reducer/nameSpace.js";
import {Genres, LoginStatus} from "../../const.js";
import MOVIES from "../../mocks/films.js";

const mockStore = configureStore([]);

const movie = {
  backgroundColor: `#D8E3E5`,
  backgroundImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Moonrise_Kingdom.jpg`,
  description: `A pair of young lovers flee their New England town, which causes a local search party to fan out to find them.`,
  director: `Wes Anderson`,
  genre: `Adventure`,
  id: 2,
  isFavorite: false,
  name: `Moonrise Kingdom`,
  posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Moonrise_Kingdom.jpg`,
  previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/moonrise-kingdom.jpg`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  rating: 7.9,
  ratingString: `good`,
  released: 2012,
  runTime: 94,
  scoresCount: 291183,
  starring: [`Jared Gilman`, `Kara Hayward`, `Bruce Willis`],
  videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
};

const isMoviePlaying = false;

it(`Проверка снепшота компонента MoviePage`, () => {
  const store = mockStore({
    [NameSpace.APP]: {
      genre: Genres.ALL,
    },
    [NameSpace.DATA]: {
      movies: MOVIES,
    },
    [NameSpace.USER]: {
      reviews: [],
    },
  });
  const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <MoviePage
            movie={movie}
            onPlayButtonHandler={() => {}}
            onExitButtonHandler={() => {}}
            isMoviePlaying={isMoviePlaying}
            authorizationStatus = {LoginStatus.NO_AUTH}
            userAvatar={``}
            onMyListClick = {() => {}}
            match={{params: {id: `1`}, isExact: true, path: ``, url: ``}}
          />
        </Provider>
      </BrowserRouter>
      , {
        createNodeMock: () => {
          return {};
        }})
    .toJSON();

  expect(tree).toMatchSnapshot();
});

