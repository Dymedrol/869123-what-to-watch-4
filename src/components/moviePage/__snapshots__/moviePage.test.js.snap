// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка снепшота компонента MoviePage 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "background": "#D8E3E5",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="The Grand Budapest Hotel"
          src="https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Moonrise_Kingdom.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head undefined"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            className="user-block__link"
            href="/login"
            onClick={[Function]}
          >
            Sign in
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Moonrise Kingdom
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Adventure
            </span>
            <span
              className="movie-card__year"
            >
              2012
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Moonrise_Kingdom.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item "
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item "
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              7.9
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                good
              </span>
              <span
                className="movie-rating__count"
              >
                94
                 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              A pair of young lovers flee their New England town, which causes a local search party to fan out to find them.
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
                Wes Anderson
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: 
                Jared Gilman, Kara Hayward, Bruce Willis
                 and other
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <div
          className="catalog__movies-list"
        >
          <article
            className="small-movie-card catalog__movies-card"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                className="player__video"
                poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/moonrise-kingdom.jpg"
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              <a
                className="small-movie-card__link"
              >
                Moonrise Kingdom
              </a>
            </h3>
          </article>
        </div>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
